### PR TITLE
fix(sdk-core): ecdsa tss signing flow update

### PR DESCRIPTION
**Pull request by olibeyene**
Ticket: BG-57413

WP updated ecdsa tss signing flow, this pr is matching that signing flow

### CODE REVIEW

**Review by byhow — APPROVED** [review]
has very little context on how TSS flow works, might be good to get someone else who's familiar with that to get a review as well

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -181,8 +184,8 @@ export async function sendShareToBitgo(
   walletId: string,
   txRequestId: string,
   shareType: SendShareType,
-  share: SShare | MUShare | KShare,
-  dShare?: DShare
+  share: SShare | MuDShare | KShare,
```
Comment (sachushaji) [inline modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:187]: NIT: can you update the function params description also with the change

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts [hunk 2]
```diff
@@ -192,27 +195,28 @@ export async function sendShareToBitgo(
 
   switch (shareType) {
     case SendShareType.KShare:
+      assert(signerShare, `signer share must be present`);
```
Comment (sachushaji) [inline modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:198]: Also I can see that signerShare also sent out in MuShare block also, so would the assert applicable for both the cases?

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 3]
```diff
@@ -290,36 +290,42 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
...
     }
+
     const MPC = new Ecdsa();
     const signingKey = MPC.keyCombine(userSigningMaterial.pShare, [
       userSigningMaterial.bitgoNShare,
       userSigningMaterial.backupNShare,
     ]);
 
     const userSignShare = await ECDSAMethods.createUserSignShare(signingKey.xShare, signingKey.yShares[3]);
+    const signerShare = userSigningMaterial.bitgoNShare.u + userSigningMaterial.bitgoNShare.chaincode;
```
Comment (sachushaji) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:301]: Why are we sending userSigning material encrypted if its available locally while user signs it? unless I am missing something

**Review by sachushaji — APPROVED** [review]
Some NITs and Question, otherwise LGTM

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts (hunk 3 above)
Comment (olibeyene) [reply modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:301]: @johnoliverdriscoll can you explain why hsm needs the user's signing material? I'm not sure why there's a need but the first call to hsm requires that

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts (hunk 2 above)
Comment (olibeyene) [reply modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:198]: signerShare is only required when sending K share i.e. the first part of the signing. at other stage signer share could be whatever.

File: modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts (hunk 1 above)
Comment (olibeyene) [reply modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts:187]: we can but I don't want to hold up this pr

**Pull request merged**